testplan/web_ui/testing/src/state/store.js

```diff
@@ -0,0 +1,127 @@
+import { configureStore } from '@reduxjs/toolkit/dist/redux-toolkit.esm';
+import { combineReducers } from '@reduxjs/toolkit/dist/redux-toolkit.esm';
+import { getDefaultMiddleware } from '@reduxjs/toolkit/dist/redux-toolkit.esm';
+import { setUseProxies } from 'immer/dist/immer.esm';
+import { fetchReport } from '../Report/BatchReportBeta/state/reportActions';
+// eslint-ignore-next-line max-len
+import { getLastResponseContentLength } from '../Report/BatchReportBeta/state/reportSelectors';
+import { setSelectedEntry } from '../Report/BatchReportBeta/state/uiActions';
+import uiHistory from '../Report/BatchReportBeta/state/uiHistory';
+import appHistory from './appHistory';
+import { flattened } from '../Common/utils';
+
+// At @reduxjs/toolkit/dist/redux-toolkit.esm.js:1422 (@reduxjs/toolkit@1.3.4)
+// ES5 compatibility is enabled, which switches off Proxy support, which is bad:
+// https://immerjs.github.io/immer/docs/performance#:~:text=The%20ES5%20fallback%20implementation
+setUseProxies(true);
+
+const __DEV__ = process.env.NODE_ENV !== 'production';
+const FIX_DEVTOOL_SPEED = process.env.REACT_APP_FIX_DEVTOOL_SPEED === 'true';
+// eslint-disable-next-line max-len
+const DISABLE_REDUX_DEVTOOLS = process.env.REACT_APP_DISABLE_REDUX_DEVTOOLS === 'true';
+let shaveObjectEntriesFromObject = null;
+if(__DEV__ && FIX_DEVTOOL_SPEED && !DISABLE_REDUX_DEVTOOLS) {
+  console.warn(flattened`
+    You've set the environment variable REACT_APP_FIX_DEVTOOL_SPEED=true
+    which will disable several features in Redux Devtools.
+  `);
+  const _isObjectLike = require('lodash/isObjectLike');
+  const _isFunction = require('lodash/isFunction');
+  shaveObjectEntriesFromObject = (entry, placeholder = '<<omitted>>') => (
+    !_isObjectLike(entry) ? entry : Object.fromEntries(
+      Object.entries(entry).map(([ prop, val ]) => [
+        prop, _isObjectLike(val)
+          ? _isFunction(placeholder) ? placeholder(prop, val) : placeholder
+          : val
+      ])
+  ));
+}
+
+const createReducer = () => combineReducers({
+  app: require('./appSlice').default.reducer,
+  // eslint-disable-next-line max-len
+  report: require('../Report/BatchReportBeta/state/reportSlice').default.reducer,
+  ui: require('../Report/BatchReportBeta/state/uiSlice').default.reducer,
+});
+
+const store = configureStore({
+  reducer: createReducer(),
+  middleware: [
+    ...getDefaultMiddleware({
+      thunk: {
+        extraArgument: { uiHistory, appHistory }
+      },
+      serializableCheck: __DEV__ && (!FIX_DEVTOOL_SPEED ? true : {
```

Comment:
I found this !cond ? s1: s2 not easy to read. Also do we want to disable these checks under production mode?

Reply:
these checks are purely for development